### DISCUSSION TITLE
是否能在持续写入TsFile文件的同时对其进行查询呢？

### DISCUSSION THREAD

**Discussion by AfterTheStorm** [apache/iotdb#8062] [Q&A]
如题，想实现持续写入TsFile文件的同时，也能支持查询。但是似乎只有执行TsFileWriter的close()方法后，才能对相应文件进行查询？执行close()方法后，似乎无法再向该文件写入新数据。

**Comment by HTHou** [comment]
是的，TsFile只有在close之后才能用现有查询接口进行查询。

close 后如果想继续向文件里追加写入，需要用 ForceAppendTsFileWriter 重新打开文件进行写入。可参考 example/tsfile 里的 TsFileForceAppendWrite

**Reply by AfterTheStorm** [reply]
了解了，非常感谢！
TsFile模块后续是否会考虑优化支持这一场景呢？目前来看，如果要实现该场景的话，在查询时得先close文件，调用查询接口进行查询，之后再重新打开文件重新写入，等于是查询干预了写入流程。除此之外似乎只能在内存中再维护一套查询模块。

**Reply by HTHou** [reply]
如果不close的话，文件尾部的索引无法生成，如果要查询的话需要从头到尾扫描数据，查询效率不会很高，所以目前没有考虑这个场景。
可以讨论一下要有什么方案可以解决这个问题。

TsFile设计本身就是以一种不可修改的形式。在IoTDB中也只有在做文件合并的时候才会对TsFile做修改操作。
